core/domain/blog_services.py

```diff
@@ -949,3 +965,70 @@ def get_blog_post_ids_matching_query(
             'Could not fulfill search request for query string %s; at least '
             '%s retries were needed.' % (query_string, MAX_ITERATIONS))
     return (valid_blog_post_ids, search_offset)
+
+
+def create_blog_author_details_model(user_id: str) -> None:
+    """Creates a new blog author details model.
+
+    Args:
+        user_id: str. The user ID of the blog author.
+    """
+    user_settings = user_services.get_user_settings(user_id, strict=True)
+    # Adding an if statement for mypy type checks to pass.
+    if user_settings.username:
+        blog_models.BlogAuthorDetailsModel.create(
+            user_id,
+            user_settings.username,
+            user_settings.user_bio
+        )
+
+
+def get_blog_author_details(user_id: str) -> Optional[
+    blog_domain.BlogAuthorDetails
+]:
+    """Returns the blog author details for the given user id. If
+    blogAuthorDetailsModel is not present, a new model with default values is
+    created.
+
+    Args:
+        user_id: str. The user id of the blog author.
+
+    Returns:
+        BlogAuthorDetails. The blog author details for the given user id.
```

Comment:
Can you please update the doc in what case None value is expected from this function? I'm asking because Optional is mentioned in typing.

Reply:
Removed `Optional[..]`